src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs

```diff
@@ -124,9 +126,18 @@ public async IAsyncEnumerable<JsonRpcResult> ProcessAsync(PipeReader reader, Jso
 
         // Handles general exceptions during parsing and validation.
         // Sends an error response and stops the stopwatch.
-        JsonRpcResult GetParsingError(string error, Exception? exception = null)
+        JsonRpcResult GetParsingError(ReadOnlySequence<byte> buffer, string error, Exception? exception = null)
         {
             Metrics.JsonRpcRequestDeserializationFailures++;
+
+            if (Encoding.UTF8.TryGetString(buffer, out string data))
+            {
+                const int sliceSize = 1000;
+                error = data.Length > sliceSize
```

Comment:
Can potentially show request data at the DEBUG level instead, to avoid overflowing the logs with it.

Reply:
Parse whole string, to then slice it to 1000 is not great for performance, first slice it to some length, then decode.

Reply:
Yeah would also do it in debug only.

Reply:
> Parse whole string, to then slice it to 1000 is not great for performance, first slice it to some length, then decode.

Yeah, that's why [this](https://github.com/NethermindEth/nethermind/pull/7621/commits/f9c0237fde167c7e175e047e01fbb3035a70effb#diff-0bf00aa62ebcd0370f2cc4c8e650028e8905fd9e4d960a5f5829d878fd0feca8) was my initial implementation - linear, almost without redundant allocations.

Problem is, it only works with `ReadOnlySpan` and here we have `ReadOnlySequence`, which causes several problems:
- `Encoding` does not support parsing sequence to a pre-allocated array, as for span.
  - It's possible to replicate this via its `Decoder` and a custom cycle, but this will look complex and will probably have worse performance.
  - Another option would be to get span from the sequence, but that would mean additional data copying if latter has multiple segments.
- UTF8 character size varies. While for regular RPC JSON all would probably use a single byte, slicing bytes may break decoding if any irregular character is involved.

I'll try to find best compromise.

Reply:
Maybe just use `Encoding.TryGetChars`?

Reply:
- It doesn't support `ReadOnlySequence`, so  copying may be needed if it has multiple segments.
- It just returns `false` if `chars` array is not enough.
![image](https://github.com/user-attachments/assets/53d027d0-f9ca-4c3e-9c92-636e493fc63c)

Reply:
Ok, maybe instead of `1000` chars, let's just do `bytes.FirstSpan`, that should be cheap?

Reply:
Tried to solve this without copying a lot of framework code.
For now method will:
- do a simple call to `Decoder.Convert` when sequence has a single segment (probably most, if not all cases)
- call `GetChars` extension method and catch a specific kind of `ArgumentException` to detect chars overflow, if multiple segments are present

Both options are covered by unit tests.

If the second case still raises concerns, I'll do with a modified internal code.

Reply:
Would be good to explain that in comments in detail